tokio-trace/src/span.rs

```diff
@@ -292,7 +292,7 @@ impl<'a> Span<'a> {
     /// Returns the result of evaluating `f`.
     pub fn enter<F: FnOnce() -> T, T>(&mut self, f: F) -> T {
         match self.inner.take() {
-            Some(inner) => dispatcher::with_default(inner.subscriber.clone(), || {
+            Some(inner) => dispatcher::with_default(&inner.subscriber.clone(), || {
```

Comment:
@carllerche/@seanmonstar, this is why I didn't originally have `with_default` take the dispatcher by ref. We have to clone here anyway, since otherwise we can't move `inner` into the closure as it's still borrowed. So, I didn't want to clone the `Arc` twice when it was really only necessary to do so a single time...